### PR TITLE
Add functions to set chunk status

### PR DESCRIPTION
**Pull request by svenklemm**
Add 2 helper functions to modify status of a chunk
ts_chunk_add_status(Chunk *chunk, int status);
ts_chunk_set_status(Chunk *chunk, int status);

### CODE REVIEW

**Review by gayyappan — APPROVED** [review]

File: src/chunk.c [hunk 1]
```diff
@@ -2985,6 +3023,20 @@ ts_chunk_set_schema(Chunk *chunk, const char *newschema)
 	return chunk_update_form(&chunk->fd);
 }
 
+bool
+ts_chunk_add_status(Chunk *chunk, int status)
```
Comment (mfundul) [inline src/chunk.c:3027]:
```suggestion
ts_chunk_add_status(Chunk *chunk, int32 status)
```
This should match `FormData_chunk`.

File: src/chunk.c [hunk 2]
```diff
@@ -2985,6 +3023,20 @@ ts_chunk_set_schema(Chunk *chunk, const char *newschema)
...
+{
+	return ts_chunk_set_status(chunk, chunk->fd.status | status);
+}
+
+bool
+ts_chunk_set_status(Chunk *chunk, int status)
```
Comment (mfundul) [inline src/chunk.c:3033]:
```suggestion
ts_chunk_set_status(Chunk *chunk, int32 status)
```
This should match `FormData_chunk`.

File: src/chunk.h [hunk 3]
```diff
@@ -138,6 +138,8 @@ extern TSDLLEXPORT void ts_chunk_drop_fks(Chunk *const chunk);
 extern TSDLLEXPORT void ts_chunk_create_fks(Chunk *const chunk);
 extern int ts_chunk_delete_by_hypertable_id(int32 hypertable_id);
 extern int ts_chunk_delete_by_name(const char *schema, const char *table, DropBehavior behavior);
+extern TSDLLEXPORT bool ts_chunk_add_status(Chunk *chunk, int status);
+extern TSDLLEXPORT bool ts_chunk_set_status(Chunk *chunk, int status);
```
Comment (mfundul) [inline src/chunk.h:142]:
```suggestion
extern TSDLLEXPORT bool ts_chunk_add_status(Chunk *chunk, int32 status);
extern TSDLLEXPORT bool ts_chunk_set_status(Chunk *chunk, int32 status);
```
same as above.

File: src/chunk.c [hunk 4]
```diff
@@ -2985,6 +3023,20 @@ ts_chunk_set_schema(Chunk *chunk, const char *newschema)
 	return chunk_update_form(&chunk->fd);
 }
 
+bool
+ts_chunk_add_status(Chunk *chunk, int status)
+{
+	return ts_chunk_set_status(chunk, chunk->fd.status | status);
```
Comment (mfundul) [inline src/chunk.c:3029]:
```suggestion
	return ts_chunk_set_status(chunk, ((uint32)chunk->fd.status) | (uint32)status);
```
Bitwise operators on signed operands produce implementation specific or undefined behaviour, so we must cast to unsigned first.

**Review by mfundul — CHANGES_REQUESTED** [review]

**Review by mfundul — APPROVED** [review]

**Pull request merged**